packages/flutter/lib/src/material/dropdown_theme.dart

```diff
@@ -0,0 +1,238 @@
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+///  * [DropdownTheme] which overrides the default [DropdownTheme]
+///    of its [DropdownButton] and [DropdownButtonFormField] descendants.
+///  * [ThemeData.textTheme], text with a color that contrasts with the card
+///    and canvas colors.
+///  * [ThemeData.colorScheme], the thirteen colors that most Material widget
+///    default colors are based on.
+@immutable
+class DropdownThemeData with Diagnosticable {
+  /// Creates a [DropdownThemeData].
+  const DropdownThemeData ({
+    this.dropdownColor,
+    this.style,
+    this.iconColor,
+    this.focusColor,
+    this.menuMaxHeight,
+    this.borderRadius,
+    this.inputDecorationTheme,
+  });
+
+  /// Overrides the default value of [DropdownButton.dropdownColor] and
+  /// [DropdownButtonFormField<T>.dropdownColor].
+  final Color? dropdownColor;
+
+  /// Overrides the default value of [DropdownButton.style] and
+  /// [DropdownButtonFormField<T>.style].
+  final TextStyle? style;
+
+  /// Overrides the default value of [DropdownButton.iconEnabledColor],
+  /// [DropdownButton.iconDisabledColor], [DropdownButtonFormField<T>.iconEnabledColor]
+  /// and [DropdownButtonFormField<T>.iconDisabledColor].
+  final MaterialStateProperty<Color?>? iconColor;
+
+  /// Overrides the default value of [DropdownButton.focusColor] and
+  /// [DropdownButtonFormField<T>.focusColor].
+  final Color? focusColor;
+
+  /// Overrides the default value of [DropdownButton.menuMaxHeight] and
+  /// [DropdownButtonFormField<T>.menuMaxHeight].
+  final double? menuMaxHeight;
+
+  /// Overrides the default value of [DropdownButton.borderRadius] and
+  /// [DropdownButtonFormField<T>.borderRadius].
+  final BorderRadius? borderRadius;
+
+  /// Overrides the default input decoration theme for [DropdownButtonFormField].
+  ///
+  /// If this is null, [DropdownButtonFormField] provides its own default.
+  final InputDecorationTheme? inputDecorationTheme;
+
+  /// Creates a copy of this object with the given fields replaced with the
+  /// new values.
+  DropdownThemeData copyWith({
+    Color? dropdownColor,
+    TextStyle? style,
+    MaterialStateProperty<Color?>? iconColor,
+    Color? focusColor,
+    double? menuMaxHeight,
+    BorderRadius? borderRadius,
+    InputDecorationTheme? inputDecorationTheme,
+  }) {
+    return DropdownThemeData(
+      dropdownColor: dropdownColor ?? this.dropdownColor,
+      style: style ?? this.style,
+      iconColor: iconColor ?? this.iconColor,
+      focusColor: focusColor ?? this.focusColor,
+      menuMaxHeight: menuMaxHeight ?? this.menuMaxHeight,
+      borderRadius: borderRadius ?? this.borderRadius,
+      inputDecorationTheme: inputDecorationTheme ?? this.inputDecorationTheme,
+    );
+  }
+
+  /// Linearly interpolate between DropdownThemeData objects.
+  static DropdownThemeData? lerp(DropdownThemeData? a, DropdownThemeData? b, double t) {
```

Comment:
Why can this return null? I have looked at a few different examples like the switch theme and checkbox theme and they do not do this. However, I do know there is some variance between theme classes based on new and old preferred patterns. Was there a class you were using as a model here?

Reply:
Oh I just saw ButtonStyle is similar, that is certainly newer. I wonder if there is a chance to create more consistency in these theme patterns. (Likely a separate issue :) )

Reply:
We are definitely not consistent about the return value when both parameters are null. Returning a non-null theme data value in that case does seem odd though. We should be (become) consistent.